rust/ip-packet/src/lib.rs

```diff
@@ -1409,9 +1410,9 @@ impl<'a> IpPacket<'a> {
     }
 
     /// Unwrap this [`IpPacket`] as a DNS message, panicking in case it is not.
-    pub fn unwrap_as_dns(&self) -> hickory_proto::op::Message {
+    pub fn unwrap_as_dns(&self) -> Message<Vec<u8>> {
```

Comment:
This is only used in `make` now, I think, can we make this private?

Reply:
This is not very important since it's only used for tests, but isn't it a bit clearer if we make this?

```suggestion
    pub fn unwrap_as_dns(&'a self) -> Message<&'a [u8]> {
```

and let the caller clone

Reply:
Unfortunately, that doesn't work because `payload` doesn't track the lifetime of the underlying buffer. See https://github.com/libpnet/libpnet/issues/319.